contracts/test/fixture/_pool_tilt.js

```diff
@@ -0,0 +1,142 @@
+const hre = require("hardhat");
+const { ethers, waffle } = hre;
+const { BigNumber } = ethers;
+const { impersonateAndFundContract, mintWETH } = require("./_fixture");
+const addresses = require("../../utils/addresses");
+
+const { ousdUnits } = require("../helpers");
+
+const { factoryCreatePool } = require("./_pools");
+
+const attackerAddress = "0xd8dA6BF26964aF9D7eEd9e03E53415D37aA96045";
+let sAttacker;
+let pool;
+
+async function getPoolConfiguration(fixture, pool) {
+  const { rEthBPT, sfrxETHwstETHrEthBPT } = fixture;
+
+  if (!!rEthBPT && pool.address == rEthBPT.address) {
+    return {
+      config: {
+        platform: "balancer",
+        poolType: "metaStable",
+        poolId: await fixture.balancerREthStrategy.balancerPoolId(),
+        pool: rEthBPT,
+        // important for Balancer type pools order of these must match the Balancer pool
+        assetAddressArray: [fixture.reth.address, fixture.weth.address],
+        bptToken: rEthBPT,
+        balancerVault: fixture.balancerVault,
+      },
+    };
+  } else if (
+    !!sfrxETHwstETHrEthBPT &&
+    pool.address == sfrxETHwstETHrEthBPT.address
+  ) {
+    return {
+      config: {
+        platform: "balancer",
+        poolType: "composableStable",
+        poolId: await fixture.balancerSfrxWstRETHStrategy.balancerPoolId(),
+        pool: sfrxETHwstETHrEthBPT,
+        // important for Balancer type pools order of these must match the Balancer pool
+        assetAddressArray: [
+          sfrxETHwstETHrEthBPT.address,
+          addresses.mainnet.wstETH,
+          addresses.mainnet.sfrxETH,
+          addresses.mainnet.rETH,
+        ],
+        bptToken: sfrxETHwstETHrEthBPT,
+        balancerVault: fixture.balancerVault,
+      },
+    };
+  }
+}
+
+async function _fundAttackerOption({
+  asset, // attacking asset
+  assetAmount, // required funded asset amount to perform tilt
+  fixture,
+}) {
+  const attackerEthBalance = await waffle.provider.getBalance(attackerAddress);
+  const assetBalance = await asset.balanceOf(attackerAddress);
+
+  if (
+    attackerEthBalance.gte(ousdUnits("100000")) ||
+    assetBalance.gte(assetAmount)
+  ) {
```

Comment:
This is confusing and feels like a bug. If account `a` has 100k ETH but wants 100 frxETH for a test case, this function won't fund them with sfrxETH?

Should this be `&&` instead of `||`? If `&&`, why check ETH balance?

Reply:
nice catch thanks yes this is a bug, should definitely be a `&&`.
My thought process is that accounts require ETH & the required asset to function. So either is missed we re-fund it.

Reply:
nit: I'm just thinking that it's probably better to split this into two conditional statements:

```
if (attackerEthBalance.lte(ousdUnits("100"))) {
// Fund it with ETH
}

if (assetBalance < xxx) {
// Fund with asset
}
```

Also, checking if ETH balance is over `ousdUnits("100000")` seems like an overkill, probably better to just check if it has 1 or 2 ETH as well 😅

Reply:
the 1m eth check is there to see if this function has funded it. Later in the function WETH is minted and that requires 500k ETH on the wallet. 

I've made this more efficient like you've suggested and also fund a bug where if accounts would have already been funded a signer wasn't returned.